### ISSUE TITLE
Build: Access variables from other components

### ISSUE THREAD

**Issue by DevVersion** [angular/material#9196]
It is currently not possible to access variables from other components because of the module compilation.

It would be helpful to be able to access variables from other components.

e.g  #9194

**Comment by ThomasBurleson** [comment]
BTW - The only reasonable option is to **share state**...

**Comment by DevVersion** [comment]
Not sure I understand what you mean. It's obvious that we can't easily import the other component files, because it would generate the unnecessary CSS twice. 

But the idea is, to only share the variables between the components.

**Comment by DevVersion** [comment]
So I took at some `node-sass` options and issues. 

Accessing variables from other components works for the docs site, because we're concatenating all SCSS files and then run the sass compiler.

When compiling a standalone module, we are not able to access any other variable, because we only have the single SCSS file as a source.

I played a bit with the "underscore" concept of SASS, to import a file without compiling it actually. But this approach didn't work, because we don't want to prefix each SCSS file with an underscore.

I also took a look at plugins like `import-resolve` and `sass-variables-loader`, but those ideas are not pretty convenient. 

I'm closing this issue, since it is unfortunately not possible to pass an array of files, to be used as resources for compilation.
